### PR TITLE
Add vamp-plugin-sdk

### PR DESCRIPTION
**Pull request by pfultz2**
This add the vamp plugin sdk.

### CODE REVIEW

File: Library/Formula/vamp-plugin-sdk.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (asparagui) [inline Library/Formula/vamp-plugin-sdk.rb]: strip this

File: Library/Formula/vamp-plugin-sdk.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class VampPluginSdk < Formula
+  homepage "http://www.vamp-plugins.org/"
+  url "http://sourceforge.net/projects/vamp/files/vamp-plugin-sdk/2.2.1/vamp-plugin-sdk-2.2.1.tar.gz/download"
+  version "2.2.1"
+  sha256 "571481098270133d2b78c6a461b850e04a98ab38284227c4d8056385f6333c26"
+
+  depends_on 'pkg-config' => :build
+  depends_on 'libsndfile'
+
+  def patches
```
Comment (DomT4) [inline Library/Formula/vamp-plugin-sdk.rb]:
```ruby
patch do
  url "https://gist.github.com/raw/462794/2f99c9305a77893974d97045860d8e61fb4d0c40/osx_settings"
  sha256 ""
end
```

And so on.

**Comment by DomT4** [comment]
Lots of comments from the audit on this: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24594/version=mavericks/testReport/junit/brew-test-bot/mavericks/audit_vamp_plugin_sdk___strict/

File: Library/Formula/vamp-plugin-sdk.rb (hunk 2 above)
Comment (pfultz2) [inline Library/Formula/vamp-plugin-sdk.rb]: Homebrew doesn't execute the patches in the correct order using the `patch do`.

Comment (DomT4) [inline Library/Formula/vamp-plugin-sdk.rb]: In which way?

You control the order with whatever order you put the ` patch do ` block in. The first patch will be processed and applied first, etc.

Comment (pfultz2) [inline Library/Formula/vamp-plugin-sdk.rb]: It seems to apply both patches at the same time. I was doing this:
```ruby
patch do
  url "https://gist.github.com/raw/462794/2f99c9305a77893974d97045860d8e61fb4d0c40/osx_settings"
  sha256 ""
end

patch do
  url "http://gist.github.com/raw/462794/86ae51b2e94b075aa2652869749e6b660badf0c1/fix_library_paths"
  sha256 ""
end
```

Is that way to do two patches, with two do blocks?

Comment (DomT4) [inline Library/Formula/vamp-plugin-sdk.rb]: You may need to do a ` --verbose ` build to see the patch detail. A normal brew installation only provides the bare minimum detail. Yup, that'd be the way - With SHA256 added of course :wink:

Comment (pfultz2) [inline Library/Formula/vamp-plugin-sdk.rb]: This is the output I get from the `patch do` block:

```
==> Patching
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/vamp-plugin-sdk--patch20150421-49990-1kpu654/osx_settings
patching file Makefile.in
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/vamp-plugin-sdk--patch20150421-49990-zjta5v/fix_library_paths
patching file Makefile.in
Hunk #1 FAILED at 102.
1 out of 1 hunk FAILED -- saving rejects to file Makefile.in.rej
Error: Failure while executing: /usr/bin/patch -g 0 -f -p1 -i /private/tmp/vamp-plugin-sdk--patch20150421-49990-zjta5v/fix_library_paths
```

If I use the `def patches` I don't get this error.

Comment (DomT4) [inline Library/Formula/vamp-plugin-sdk.rb]: I believe ` def patches ` is deprecated. Could be wrong. @mikemcquaid?

**Comment by pfultz2** [comment]
> I believe def patches is deprecated. Could be wrong. @mikemcquaid?

It probably is, however, the `patch do` is partially broken, so I don't think it is a full replacement for `patches` yet.

**Comment by pfultz2** [comment]
Ok so I combine them into one patch.

File: Library/Formula/vamp-plugin-sdk.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+
+class VampPluginSdk < Formula
+  homepage "http://www.vamp-plugins.org/"
+  url "https://sourceforge.net/projects/vamp/files/vamp-plugin-sdk/2.2.1/vamp-plugin-sdk-2.2.1.tar.gz"
+  sha256 "571481098270133d2b78c6a461b850e04a98ab38284227c4d8056385f6333c26"
+
+  depends_on "pkg-config" => :build
+  depends_on "libsndfile"
+
+  patch :p1, :DATA
```
Comment (mikemcquaid) [inline Library/Formula/vamp-plugin-sdk.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/vamp-plugin-sdk.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  def install
+    system "./configure", "--disable-debug", "--disable-dependency-tracking", "--prefix=#{prefix}"
+    system "make", "install"
+  end
+end
+
```
Comment (mikemcquaid) [inline Library/Formula/vamp-plugin-sdk.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by DomT4** [comment]
Mike's feedback here needs addressing, Thanks!

**Comment by DomT4** [comment]
Closing due to no response. If you're still interested please fix the comments above and request the PR be reopened, Thanks.

**Pull request closed without merging**